src/Runner.Worker/JobExtension.cs

```diff
@@ -248,6 +249,19 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                     Trace.Info("Downloading actions");
                     var actionManager = HostContext.GetService<IActionManager>();
                     var prepareResult = await actionManager.PrepareActionsAsync(context, message.Steps);
+
+                    // add hook to preJobSteps
+                    var startedHookPath = Environment.GetEnvironmentVariable("ACTIONS_RUNNER_HOOK_JOB_STARTED");
+                    if (!string.IsNullOrEmpty(startedHookPath))
+                    {
+                        var hookStep = new ManagedScriptStep(startedHookPath,
+                                                                $"{PipelineTemplateConstants.Always}()",
+                                                                displayName: "Set up runner",
+                                                                ActionRunStage.Pre);
+                        hookStep.Initialize(HostContext); // TODO: is Initialize() the right way to pass on a HostContext?
```

Comment:
i think you should follow other code for creating `ManagedScriptStep`, ex: `HostContext.CreateService<IManagedScriptStep>()` similar to `var actionRunner = HostContext.CreateService<IActionRunner>();`

Reply:
Sounds good! Should we apply the same logic to `new JobExtensionRunner` at line 278?